test/spec/services/limitsRangesServiceSpec.js

```diff
@@ -0,0 +1,401 @@
+      var containers = [{
+        name: 'my-container',
+        resources: {
+          requests: {
+            memory: '2Gi'
+          }
+        }
+      }];
+
+      var memoryProblems = LimitRangesService.validatePodLimits(mockLimitRanges, 'memory', containers, mockProject);
+      expect(memoryProblems.length).toBeGreaterThan(0);
+    });
+
+    it("should report problems when memory request is too low", function() {
+      var containers = [{
+        name: 'my-container',
+        resources: {
+          requests: {
+            memory: '50Mi'
+          }
+        }
+      }];
+
+      var memoryProblems = LimitRangesService.validatePodLimits(mockLimitRanges, 'memory', containers, mockProject);
+      expect(memoryProblems.length).toBeGreaterThan(0);
+    });
+
+    it("should report problems when memory limit is too high", function() {
+      var containers = [{
+        name: 'my-container',
+        resources: {
+          limits: {
+            memory: '2Gi'
+          }
+        }
+      }];
+
+      var memoryProblems = LimitRangesService.validatePodLimits(mockLimitRanges, 'memory', containers, mockProject);
+      expect(memoryProblems.length).toBeGreaterThan(0);
+    });
+
+    it("should report problems when memory limit is too low", function() {
+      var containers = [{
+        name: 'my-container',
+        resources: {
+          limits: {
+            memory: '50Mi'
+          }
+        }
+      }];
+
+      var memoryProblems = LimitRangesService.validatePodLimits(mockLimitRanges, 'memory', containers, mockProject);
+      expect(memoryProblems.length).toBeGreaterThan(0);
+    });
+
+    it("should report problems when cpu request is too high", function() {
+      var containers = [{
+        name: 'my-container',
+        resources: {
+          requests: {
+            cpu: '3'
+          }
+        }
+      }];
+
+      var cpuProblems = LimitRangesService.validatePodLimits(mockLimitRanges, 'cpu', containers, mockProject);
+      expect(cpuProblems.length).toBeGreaterThan(0);
+    });
+
+    it("should report problems when cpu request is too low", function() {
+      var containers = [{
+        name: 'my-container',
+        resources: {
+          requests: {
+            cpu: '10m'
+          }
+        }
+      }];
+
+      var cpuProblems = LimitRangesService.validatePodLimits(mockLimitRanges, 'cpu', containers, mockProject);
+      expect(cpuProblems.length).toBeGreaterThan(0);
+    });
+
+    it("should report problems when cpu limit is too high", function() {
+      var containers = [{
+        name: 'my-container',
+        resources: {
+          limits: {
+            cpu: '3'
+          }
+        }
+      }];
+
+      var cpuProblems = LimitRangesService.validatePodLimits(mockLimitRanges, 'cpu', containers, mockProject);
+      expect(cpuProblems.length).toBeGreaterThan(0);
+    });
+
+    it("should report problems when cpu limit is too low", function() {
+      var containers = [{
```

Comment:
you redefine these same sets of containers objects a bunch of times, maybe define them above as `lowCPULimit` and `highCPULimit` close to where the limit range is defined ?  And then its easier to validate the tests are right?